src/lib.rs

```diff
@@ -186,12 +234,6 @@ impl Matcher for TreeSitterMatcher<'_> {
 
     fn find_at(&self, haystack: &[u8], at: usize) -> Result<Option<Match>, Self::Error> {
         let mut matches_info = self.matches_info.borrow_mut();
-        if let Some(matches_info) = matches_info.as_ref() {
```

Comment:
We don't in fact always get called with progressively shorter trailing substrings of the file contents (per the reason `grep` was forked there are other calls for doing some kind of "re-matching" happening)